src/pages/my_page/user_profile/UserOrcid.tsx

```diff
@@ -18,11 +18,44 @@ import { UrlPathTemplate } from '../../../utils/urlPaths';
 import { authenticatedApiRequest } from '../../../api/apiRequest';
 import { getValueByKey } from '../../../utils/user-helpers';
 import { fetchPerson } from '../../../api/cristinApi';
+import { postOrcidCredentials } from '../../../api/orcidApi';
+import { OrcidCredentials } from '../../../types/orcid.types';
 
 interface UserOrcidProps {
   user: User;
 }
 
+function allPropertiesAreInCredentialsObject(rawCredentials: { [p: string]: string }) {
+  return (
+    rawCredentials.hasOwnProperty('expires_in') &&
+    isNaN(+rawCredentials['expires_in']) &&
+    rawCredentials.hasOwnProperty('id_token') &&
+    rawCredentials.hasOwnProperty('persistent') &&
+    rawCredentials.hasOwnProperty('tokenId') &&
+    isNaN(+rawCredentials['tokenId']) &&
+    rawCredentials.hasOwnProperty('tokenVersion') &&
+    rawCredentials.hasOwnProperty('token_type') &&
+    rawCredentials.hasOwnProperty('access_token')
+  );
+}
+
+const getOrcidCredentials = (search: string, orcidUrl: string): OrcidCredentials | null => {
+  const searchParams = new URLSearchParams(search);
+  const rawCredentials = Object.fromEntries(searchParams);
+  return allPropertiesAreInCredentialsObject(rawCredentials)
+    ? {
+        expiresIn: +rawCredentials['expires_in'],
+        idToken: rawCredentials['id_token'],
+        persistent: rawCredentials['persistent'].toLowerCase() == 'true',
```

Comment:
Alltid 3 `=` i JS, fordi JS :D 
```suggestion
        persistent: rawCredentials['persistent'].toLowerCase() === 'true',
```

Reply:
Fikset